internal/invocation/stripe_executor.go

```diff
@@ -0,0 +1,73 @@
+package invocation
+
+import (
+	"math/rand"
+	"sync"
+)
+
+// stripeExecutor executes given "tasks" preserving the order among the ones
+// that are given with the same key
+type stripeExecutor struct {
+	quit            chan struct{}
+	wg              *sync.WaitGroup
+	executeFunction func(queue chan func(), quit chan struct{}, wg *sync.WaitGroup)
+	tasks           []chan func()
```

Comment:
Let's call this field `taskQueues` to make it more clearer.

Reply:
Sounds great.